importlab/resolve.py

```diff
@@ -95,10 +94,57 @@ def convert_to_path(name):
     return (filename, dot_count)
 
 
+def infer_module_name(filename, fspath):
+    """Convert a python filename to a module relative to pythonpath."""
+    filename, ext = os.path.splitext(filename)
+    if not ext == '.py':
+        return ''
+    for path in fspath:
+        root = getattr(path, 'root', None)
+        if not root:
+            continue
+        if filename.startswith(root):
+            short_name = filename[len(root) + 1:]
+            return short_name.replace(os.path.sep, '.')
+    else:
+        # We have not found filename relative to anywhere in pythonpath.
+        return ''
+
+
+def get_absolute_name(parent, relative_name):
+    """Joins a parent module name and a relative name.
+
+    Args:
+      parent: A dotted name, e.g. foo.bar.baz
+      relative_name: A dotted name with possibly some leading dots, e.g. ..x.y
+
+    Returns:
+      The relative name appended to the parent's package, after going up one
+      level for each leading dot.
+        e.g. foo.bar.baz + ..hello.world -> foo.hello.world
+      relative_name if it does not start with a dot
+      None if the relative name has too many leading dots.
+    """
+    prefix = parent[:parent.rfind('.')]
+    path = prefix.split('.') if prefix else []
+    name = relative_name.lstrip('.')
+    ndots = len(relative_name) - len(name)
+    if ndots == 0 or ndots == len(path) + 1:
+        # If module foo.bar imports module ..baz.quux, return baz.quux
+        return name
+    if ndots > len(path):
+      return None
+    prefix = ''.join([p + '.' for p in path[:len(path) + 1 - ndots]])
+    return prefix + name
+
+
 class Resolver:
-    def __init__(self, fs_path, current_filename):
+    def __init__(self, fs_path, current_filename, current_module=None):
```

Comment:
I don't see 'current_module' being passed in anywhere outside of tests.

Reply:
Now it looks like current_module is always present! Does it need to be optional?

Reply:
i'm not sure; i felt that it should be possible to create a resolver even if you didn't have a module name (and it makes testing a bit easier when we don't care about that particular feature)